ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +20,107 @@ export interface DropDownState {
-        this.subscriptions = [];
-    }
+        return () => {
+            (subscriptionsRef.current || []).forEach(s => s.unsubscribe());
+            subscriptionsRef.current = [];
+        };
+    }, []);
 
-    public close() {
-        this.setState({opened: false});
-    }
-
-    private refreshState() {
-        const anchor = this.el.querySelector('.argo-dropdown__anchor') as HTMLElement;
-        const {top, left} = anchor.getBoundingClientRect();
+    const refreshState = () => {
+        const content = contentRef.current;
+        const anchor = anchorRef.current;
         const anchorHeight = anchor.offsetHeight + 2;
+        const {top: anchorTop, left: anchorLeft} = anchor.getBoundingClientRect();
+        const newState = {top: anchorTop, left: anchorLeft, opened};
 
-        const newState = {left: this.state.left, top: this.state.top, opened: this.state.opened};
         // Set top position
-        if (this.content.offsetHeight + top + anchorHeight > window.innerHeight) {
-            newState.top = top - this.content.offsetHeight - 2;
+        if (content.offsetHeight + anchorTop + anchorHeight > window.innerHeight) {
+            newState.top = anchorTop - content.offsetHeight - 2;
         } else {
-            newState.top = top + anchorHeight;
+            newState.top = anchorTop + anchorHeight;
         }
 
         // Set left position
-        if (this.content.offsetWidth + left > window.innerWidth) {
-            newState.left = left - this.content.offsetWidth + anchor.offsetWidth;
+        if (content.offsetWidth + anchorLeft > window.innerWidth) {
+            newState.left = anchorLeft - content.offsetWidth + anchor.offsetWidth;
         } else {
-            newState.left = left;
+            newState.left = anchorLeft;
         }
+
         return newState;
-    }
+    };
 
-    private open() {
-        if (!this.content || !this.el) {
+    const open = () => {
+        if (!contentRef.current || !anchorRef.current) {
             return;
         }
 
-        const newState = this.refreshState();
+        const newState = refreshState();
+
         newState.opened = true;
-        this.setState(newState);
-        dropDownOpened.next(this);
-    }
+        setOpened(newState.opened);
+        setLeft(newState.left);
+        setTop(newState.top);
+        dropDownOpened.next(DropDown);
+    };
+
+    const close = () => {
+        setOpened(false);
+    };
+
+    return (
+        <div className='argo-dropdown' ref={anchorRef}>
+            <div
+                qe-id={qeId}
+                className='argo-dropdown__anchor'
+                onClick={event => {
+                    open();
+                    event.stopPropagation();
+                }}>
+                <Anchnor />
+            </div>
+            {createPortal(
+                <div className={classNames('argo-dropdown__content', {'opened': opened, 'is-menu': isMenu})} style={{top, left}} ref={contentRef}>
```

Comment:
```suggestion
                <div className={classNames('argo-dropdown__content', {opened, 'is-menu': isMenu})} style={{top, left}} ref={contentRef}>
```
can use shorthand here since they have the same name now